--- src/main/java/com/spotify/docker/client/DefaultDockerClient.java.orig	2015-10-13 22:15:28.000000000 +0100
+++ src/main/java/com/spotify/docker/client/DefaultDockerClient.java	2016-02-10 19:57:30.637197090 +0000
@@ -795,7 +795,7 @@
       throws DockerException, InterruptedException, IOException {
     checkNotNull(handler, "handler");
 
-    WebTarget resource = resource().path("build");
+    WebTarget resource = noTimeoutResource().path("build");
 
     for (final BuildParameter param : params) {
       resource = resource.queryParam(param.buildParamName, String.valueOf(param.buildParamValue));
@@ -885,7 +885,7 @@
   @Override
   public LogStream logs(final String containerId, final LogsParameter... params)
       throws DockerException, InterruptedException {
-    WebTarget resource = resource()
+    WebTarget resource = noTimeoutResource()
         .path("containers").path(containerId).path("logs");
 
     for (final LogsParameter param : params) {
@@ -909,7 +909,7 @@
   public LogStream attachContainer(final String containerId,
                                    final AttachParameter... params) throws DockerException,
       InterruptedException {
-    WebTarget resource = resource().path("containers").path(containerId)
+    WebTarget resource = noTimeoutResource().path("containers").path(containerId)
         .path("attach");
 
     for (final AttachParameter param : params) {
